lib/utils/platform/iot_threads_afr.c

```diff
@@ -0,0 +1,350 @@
+ */
+
+/**
+ * @file iot_threads_afr.c
+ * @brief Implementation of the functions in iot_threads.h for POSIX systems.
+ */
+
+/* Build using a config header, if provided. */
+#ifdef IOT_CONFIG_FILE
+    #include IOT_CONFIG_FILE
+#endif
+
+
+/* Platform threads include. */
+#include "platform/iot_threads.h"
+#include "types/iot_platform_types.h"
+
+/* Configure logs for the functions in this file. */
+#ifdef IOT_LOG_LEVEL_PLATFORM
+    #define _LIBRARY_LOG_LEVEL        IOT_LOG_LEVEL_PLATFORM
+#else
+    #ifdef IOT_LOG_LEVEL_GLOBAL
+        #define _LIBRARY_LOG_LEVEL    IOT_LOG_LEVEL_GLOBAL
+    #else
+        #define _LIBRARY_LOG_LEVEL    IOT_LOG_NONE
+    #endif
+#endif
+
+#define _LIBRARY_LOG_NAME    ( "THREAD" )
+#include "iot_logging_setup.h"
+
+/*
+ * Provide default values for undefined memory allocation functions based on
+ * the usage of dynamic memory allocation.
+ */
+#ifndef IotThreads_Malloc
+    #include <stdlib.h>
+
+    /**
+     * @brief Memory allocation. This function should have the same signature
+     * as [malloc](http://pubs.opengroup.org/onlinepubs/9699919799/functions/malloc.html).
+     */
+    #define IotThreads_Malloc    malloc
+#endif
+#ifndef IotThreads_Free
+    #include <stdlib.h>
+
+    /**
+     * @brief Free memory. This function should have the same signature as
+     * [free](http://pubs.opengroup.org/onlinepubs/9699919799/functions/free.html).
+     */
+    #define IotThreads_Free    free
+#endif
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @defgroup Mutex types.
+ */
+/**@{ */
+#ifndef IOT_THREAD_MUTEX_NORMAL
+    #define IOT_THREAD_MUTEX_NORMAL        0                        /**< Non-robust, deadlock on relock, does not remember owner. */
+#endif
+#ifndef IOT_THREAD_MUTEX_ERRORCHECK
+    #define IOT_THREAD_MUTEX_ERRORCHECK    1                        /**< Non-robust, error on relock,  remembers owner. */
+#endif
+#ifndef IOT_THREAD_MUTEX_RECURSIVE
+    #define IOT_THREAD_MUTEX_RECURSIVE     2                        /**< Non-robust, recursive relock, remembers owner. */
+#endif
+#ifndef IOT_THREAD_MUTEX_DEFAULT
+    #define IOT_THREAD_MUTEX_DEFAULT       IOT_THREAD_MUTEX_NORMAL  /**< IOT_THREAD_MUTEX_NORMAL (default). */
+#endif
+/**@} */
+
+/*-----------------------------------------------------------*/
+
+static void _threadRoutineWrapper( void * pArgument )
+{
+    _threadInfo_t * pThreadInfo = ( _threadInfo_t * ) pArgument;
+
+    /* Run the thread routine. */
+    pThreadInfo->threadRoutine( pThreadInfo->pArgument );
+    IotThreads_Free( pThreadInfo );
+
+    vTaskDelete( NULL );
+}
+
+/*-----------------------------------------------------------*/
+
+bool Iot_CreateDetachedThread(  IotThreadRoutine_t threadRoutine,
+                                void * pArgument,
+                                int32_t priority,
+                                size_t stackSize )
+{
+    IotLogDebug( "Creating new thread." );
+    int status = 0;
```

Comment:
I think it would be simpler (and more consistent with other code) to initialize:

`bool status = true;`
and on failure:
`status = false;`
and return:
`return status;`

Reply:
Agree